### ISSUE TITLE
Panic when running clx

### ISSUE THREAD

**Issue by yngvark** [bensadeh/circumflex#2]
When running `clx` I get

```
panic: runtime error: index out of range [29] with length 29 [recovered]
	panic: runtime error: index out of range [29] with length 29

goroutine 1 [running]:
gitlab.com/tslocum/cview.(*Application).Run.func1(0xc00011fb80)
	/home/runner/go/pkg/mod/gitlab.com/tslocum/cview@v1.5.2/application.go:299 +0x87
panic(0x843d00, 0xc000a54eb8)
	/opt/hostedtoolcache/go/1.16.0/x64/src/runtime/panic.go:965 +0x1b9
clx/model.SetListItemsToCurrentPage(0xc0001d32b0, 0xc0002de0f0, 0x1d, 0x1e, 0x0, 0x1e, 0xc000020640)
	/home/runner/work/circumflex/circumflex/model/model.go:221 +0x2a5
clx/model.showPageAfterResize(0xc00007cb90, 0xc0001d32b0, 0xc000078be0, 0x4, 0x4, 0xc00010cc40, 0xc000020640)
	/home/runner/work/circumflex/circumflex/model/model.go:95 +0xa9
clx/model.SetAfterInitializationAndAfterResizeFunctions.func1(0x11a, 0x40)
	/home/runner/work/circumflex/circumflex/model/model.go:43 +0x196
gitlab.com/tslocum/cview.(*Application).Run.func3(0x80a2a0, 0xc0001d0030)
	/home/runner/go/pkg/mod/gitlab.com/tslocum/cview@v1.5.2/application.go:422 +0x31e
gitlab.com/tslocum/cview.(*Application).Run(0xc00011fb80, 0x0, 0x0)
	/home/runner/go/pkg/mod/gitlab.com/tslocum/cview@v1.5.2/application.go:457 +0x304
main.main()
	/home/runner/work/circumflex/circumflex/main.go:38 +0x150
```

Tried updating with `brew update` and running again, with the same error.

**Comment by yngvark** [comment]
When running the linux binary directly, 0.16:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x826690]

goroutine 1 [running]:
gitlab.com/tslocum/cview.(*Application).Run.func1(0xc000219540)
	/home/runner/go/pkg/mod/gitlab.com/tslocum/cview@v1.5.2/application.go:299 +0x87
panic(0x87c460, 0xbb97a0)
	/opt/hostedtoolcache/go/1.16.2/x64/src/runtime/panic.go:965 +0x1b9
clx/handler.convert(0xc0002124b0, 0x1e, 0x1e, 0x2, 0x1d, 0x1d, 0x0)
	/home/runner/work/circumflex/circumflex/handler/handler.go:124 +0x90
clx/handler.getOnlineStories(0x0, 0x0, 0x1e, 0x2, 0x101, 0xc000217cb0, 0x0, 0xc000151ce8, 0x82b04f, 0xc000184000, ...)
	/home/runner/work/circumflex/circumflex/handler/handler.go:85 +0x30b
clx/handler.(*StoryHandler).GetStories(0xc0000c25e8, 0x0, 0x0, 0x1e, 0x2, 0x48c401, 0xc000151d48, 0x1f57baec4, 0xc000151d88, 0x48c76d, ...)
	/home/runner/work/circumflex/circumflex/handler/handler.go:47 +0x11c
clx/model.SetAfterInitializationAndAfterResizeFunctions.func1(0x11a, 0x40)
	/home/runner/work/circumflex/circumflex/model/model.go:43 +0x13e
gitlab.com/tslocum/cview.(*Application).Run.func3(0x87e640, 0xc00026e000)
	/home/runner/go/pkg/mod/gitlab.com/tslocum/cview@v1.5.2/application.go:422 +0x31e
gitlab.com/tslocum/cview.(*Application).Run(0xc000219540, 0x0, 0x0)
	/home/runner/go/pkg/mod/gitlab.com/tslocum/cview@v1.5.2/application.go:457 +0x304
main.main()
	/home/runner/work/circumflex/circumflex/main.go:47 +0x17e

```

**Comment by yngvark** [comment]
Happens when I zoom out in my terminal. My default zoom level makes this crash. When zooming in it works.

**Comment by bensadeh** [comment]
Thanks for letting me know! 

I think I know where to look in order to fix this issue.

**Comment by bensadeh** [comment]
Fixed in 8e220a2f11f139f389c794f6cd58fa87151ba160.

The bug was related to an option turned on by default where `clx` hides stories of the type `Company X is hiring`. Discarding these posts while viewing the maximum items possible (30) would would throw off the internal pagination logic with off by one errors.

As a workaround `clx` now overrides and disables this option whenever the number of items to display is close to 30. 

I've created a new release with this bugfix in 0.17.
